### PR TITLE
Convert to TypeScript

### PR DESCRIPTION
**Pull request by rylanc**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
+    'typescript',
+  ],
   env: {
     'browser': true,
   },
-  rules: {},
+  settings: {
+    'import/resolver': {
+      node: {
+        extensions: ['.js', '.jsx', '.ts', '.tsx']
```
Comment (colinharris05) [inline .eslintrc.js]: Do you need `.jsx` here anymore?

File: .eslintrc.js [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
+      }
+    },
+  },
+  rules: {
+    'no-undef': 'off',
+    'typescript/no-unused-vars': 'error',
+    'import/extensions': ['error', 'always', {
+      js: 'never',
+      jsx: 'never',
```
Comment (colinharris05) [inline .eslintrc.js]: Same comment as above

File: .eslintrc.js (hunk 1 above)
Comment (rylanc) [reply .eslintrc.js]: Nope. Good catch

**Comment by rylanc** [comment]
@colinharris05 Yeah, that's the only reason for the webpack dep. One of these days, I'd like to convert the tests to jest as well

**Comment by colinharris05** [comment]
@rylanc awesome, I'd be down to help with the Jest conversion at some point so we can drop webpack and gusto karma. As long as your confident it will compile and publish fine, this looks good to me @zp-alfred accept!

**Comment by zp-alfred** [comment]
Accepted by colinharris05!

**Comment by zp-alfred** [comment]
Accepted by colinharris05!

**Pull request merged**